flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/writer/BigQueryDefaultWriter.java

```diff
@@ -47,17 +51,35 @@
  * @param <IN> Type of records to be written to BigQuery.
  */
 public class BigQueryDefaultWriter<IN> extends BaseWriter<IN> {
+    // Counter specific to at-least-once Implementation since records are written after
+    // checkpointing
+    // in Exactly Once mode.
+    Counter numberOfRecordsWrittenToBigQuerySinceCheckpoint;
 
     public BigQueryDefaultWriter(
-            int subtaskId,
             String tablePath,
             BigQueryConnectOptions connectOptions,
             BigQuerySchemaProvider schemaProvider,
-            BigQueryProtoSerializer serializer) {
-        super(subtaskId, tablePath, connectOptions, schemaProvider, serializer);
+            BigQueryProtoSerializer serializer,
+            InitContext context) {
+        super(context.getSubtaskId(), tablePath, connectOptions, schemaProvider, serializer);
         streamName = String.format("%s/streams/_default", tablePath);
         totalRecordsSeen = 0L;
         totalRecordsWritten = 0L;
+        initializeAtleastOnceFlinkMetrics(context);
+    }
+
+    /**
+     * Initialize Flink Metrics for at-least-once approach.
+     *
+     * @param context Sink Context to derive the Metric Group.ß
+     */
+    void initializeAtleastOnceFlinkMetrics(InitContext context) {
+        SinkWriterMetricGroup sinkWriterMetricGroup = context.metricGroup();
+        // Call BaseWriter's initializeMetrics() for common metrics.
+        super.initializeMetrics(sinkWriterMetricGroup);
+        this.numberOfRecordsWrittenToBigQuerySinceCheckpoint =
+                sinkWriterMetricGroup.counter("numberOfRecordsWrittenToBigQuerySinceCheckpoint");
```

Comment:
`super`/`this` are not needed for inherited methods/attributes

Reply:
Fixed.